chore: upgrade aws sdk version to 2.x

jayanta2018 opened the pull request:
Ticket TIS21-7753

jayanta2018 committed:
chore: upgrade aws sdk version to 2.x

jayanta2018 committed:
fix: fix sonar issues

JosephKelly commented on src/test/java/uk/nhs/hee/tis/revalidation/core/RevalidationCoreApplicationTest.java:
This is looks like a "Vanity Metric" antipattern to boost code coverage.  We should be hitting 80%+ without test classes like this so I'd suggest removing it.

JosephKelly replied:
If not doing that.  Please verify that the `@MockBean` has a purpose.

JosephKelly commented on src/test/java/uk/nhs/hee/tis/revalidation/core/service/AdminServiceTest.java:
```diff
@@ -65,10 +66,13 @@ void setUp() {
     // Given.
-    ListUsersInGroupResult listUsersInGroupResult = new ListUsersInGroupResult();
-    listUsersInGroupResult.setUsers(Collections.emptyList());
+    ListUsersInGroupResponse listUsersInGroupResult = ListUsersInGroupResponse.builder()
+        .build();
+    listUsersInGroupResult = listUsersInGroupResult.toBuilder().users(Collections.emptyList())
+        .build();
```
Again, condense this to use the builder pattern properly.

JosephKelly commented on src/test/java/uk/nhs/hee/tis/revalidation/core/service/AdminServiceTest.java:
```diff
@@ -80,43 +84,56 @@ void shouldReturnEmptyListWhenNoAssignableAdminsFound() {
+
+    AttributeType attributeType1 = AttributeType.builder()
+        .build();
+    attributeType1 = attributeType1.toBuilder().name("family_name").build();
+    attributeType1 = attributeType1.toBuilder().value("user1").build();
+
+    AttributeType attributeType2 = AttributeType.builder()
+        .build();
+    attributeType2 = attributeType2.toBuilder().name("family_name").build();
+    attributeType2 = attributeType2.toBuilder().value("user2").build();
+
+    AttributeType attributeType3 = AttributeType.builder()
+        .build();
+    attributeType3 = attributeType3.toBuilder().name("given_name").build();
+    attributeType3 = attributeType3.toBuilder().value("user1").build();
+
+    AttributeType attributeType4 = AttributeType.builder()
+        .build();
+    attributeType4 = attributeType4.toBuilder().name("given_name").build();
+    attributeType4 = attributeType4.toBuilder().value("user2").build();
+
+    AttributeType attributeType5 = AttributeType.builder()
+        .build();
+    attributeType5 = attributeType5.toBuilder().name("email").build();
+    attributeType5 = attributeType5.toBuilder().value("user1@email.com").build();
+
+    AttributeType attributeType6 = AttributeType.builder()
+        .build();
+    attributeType6 = attributeType6.toBuilder().name("email").build();
+    attributeType6 = attributeType6.toBuilder().value("user2@email.com").build();
+
+    user1 = user1.toBuilder()
+        .attributes(Arrays.asList(attributeType1, attributeType3, attributeType5)).build();
+    user2 = user2.toBuilder()
+        .attributes(Arrays.asList(attributeType2, attributeType4, attributeType6)).build();
+
+    ListUsersInGroupResponse listUsersInGroupResult = ListUsersInGroupResponse.builder()
+        .build();
+    listUsersInGroupResult = listUsersInGroupResult.toBuilder().users(Arrays.asList(user1, user2))
+        .build();
```
Again, condense these to use the builder pattern properly.

JosephKelly replied:
Looks like this is still outstanding.  I'd recommend reading it through and identifying it but spotted what could be a check if you are really struggling with the builder pattern.

JosephKelly replied:
I've noted what's still outstanding

JosephKelly reviewed (changes requested):
I like the exploration of using the Streams API.
Noted some things to improve.

jayanta2018 committed:
refact: handle review comments

jayanta2018 committed:
refact: handle more review comments

JosephKelly commented on src/main/java/uk/nhs/hee/tis/revalidation/core/mapper/util/AdminUtil.java:
```diff
@@ -21,54 +21,60 @@
+        .filter(attr -> "email".equals(attr.name()))
+        .map(AttributeType::value)
+        .findFirst()
+        .orElse("");
```
I was missing the return statement in reading across the diff.

JosephKelly reviewed (changes requested):
It would be good to update names in line with the types, e.g. `ListUsersInGroupResponse listUsersResult` is no longer a "result" but a "response".  The `identityProvider` could be changed too.
Please look at that in both tests and prod classes.

jayanta2018 committed:
refact: change as per review comments

JosephKelly commented on src/test/java/uk/nhs/hee/tis/revalidation/core/service/AdminServiceTest.java:
```diff
@@ -80,43 +84,56 @@ void shouldReturnEmptyListWhenNoAssignableAdminsFound() {
-    listUsersInGroupResult.setUsers(Arrays.asList(user1, user2));
-
-    when(identityProvider.listUsersInGroup(any())).thenReturn(listUsersInGroupResult);
+    UserType user1 = UserType.builder()
+        .username("user1")
+        .build();
+
+    UserType user2 = UserType.builder()
+        .username("user2")
+        .build();
+
```
You can move these as well to use a builder once for each object.

jayanta2018 committed:
refact: bring builders under method

JosephKelly committed:
refactor(test): use UserType builder once

This commit continues to use the externalised builder method.

TIS21-7753: Upgrade AWS SDK v1->v2
